db.js

```diff
@@ -1,28 +1,170 @@
-const mysql = require('mysql');
-const dotenv = require('dotenv');
+const mysql = require("mysql");
+const dotenv = require("dotenv");
 let instance = null;
 dotenv.config();
 
 // Create Database Connection
-const db = mysql.createConnection({
-    host: process.env.DB_HOST,
-    user: process.env.DB_USER,
-    password: process.env.DB_PASSWORD,
-    database: process.env.MySQL_DB
+const connection = mysql.createConnection({
+  host: process.env.DB_HOST,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+  database: process.env.MySQL_DB,
 });
 
 // To Connect
-db.connect((err) => { 
-    if (err) { 
-        throw err;
-    }
-    console.log('Connection Successful!');
+connection.connect((err) => {
+  if (err) {
+    throw err;
+  }
+  console.log("Connection Successful!");
 });
 
 class DbService {
-    static getDbServiceInstance() {
-        return instance ? instance : new DbService();
+  static getDbServiceInstance() {
+    if (!instance) instance = new DbService();
+    return instance;
+  }
+
+  joinQuizTableQuery() {
+    const joinTableQuery =
+      "SELECT * FROM quiz INNER JOIN quiz_question ON quiz.quizId = quiz_question.quizId";
+    return joinTableQuery;
+  }
+
+  intFormatter(id) {
+    const int = parseInt(id, 10);
+    return int;
+  }
+
+  async getAllQuizzes() {
+    try {
+      return new Promise((resolve, reject) => {
+        const query = "SELECT * FROM quiz;";
+
+        connection.query(query, (err, result) => {
+          if (!err) resolve(result);
+          else reject(err.message);
```

Comment:
(personal preference) I typically like to handle the "alternative" flow first, before handling the happy flow. 

i.e. 

```suggestion
          if (err) return reject(result);
          resolve (err.message);
```

Rationale: I like alternative flow to be indented. Whereas happy flow on the "top-level" indentation